### PR TITLE
DP-235: low-confidence fallback and clarification prioritization

### LINKED ISSUE #235: Add low-confidence policy fallback and clarification prioritization
> Source catalog: docs/roadmap/v0.8.0-dynamic-percolation-issues.md

# Add low-confidence policy fallback and clarification prioritization

1. Title
- Add low-confidence policy fallback and clarification prioritization

2. Parent epic
- #225

3. Problem
- low-confidence behavior exists in summary but is not first-class in policy selection.

4. Why this matters
- low confidence should suppress interruption and elevate safe clarification.

5. Proposal
- introduce explicit low-confidence policy branch and clarification candidate type.

6. Implementation notes
- likely touched: `src/summary.ts`, `src/extension.ts`, `src/percolation/ranking.ts`
- candidate action examples: open last edit, add checkpoint note, run safe validation

7. Acceptance criteria
- low-confidence state avoids notification by default
- top card promotes clarification, not speculative action

8. Test plan
- unit tests for low-confidence branch and fallback ranking

9. Telemetry / metrics impact
- add `lowConfidenceClarificationRate`

10. Risks / edge cases
- excessive fallback can feel repetitive

11. Dependencies
- DP-102, DP-104

12. Suggested labels
- `area:policy`, `post-v0.8`

13. Suggested milestone
- `post-v0.8-experiments`

14. Mockup reference
- `docs/ux/dynamic-percolation-mockups.md#default-panel-wireframe-prioritized`
## Linked Epic
- #225

### PR DESCRIPTION
**Pull request by jkordish**
## Summary
- add a first-class percolation `clarification` candidate for low-confidence summaries
- prioritize clarification in ranking and propagate the new kind to memory/explainability
- suppress non-manual low-confidence notifications by default to avoid speculative interruption
- add low-confidence clarification telemetry (`lowConfidenceClarificationRate`) to metrics schema + CSV export
- extend tests for percolation types/ranking/suppression and metrics

## Verification
- npm run compile
- npm run lint
- npm test

Closes #235

### CODE REVIEW

File: test/metrics.test.ts [hunk 1]
```diff
@@ -74,6 +74,17 @@ describe('hasAnyRecordedMetric', () => {
     expect(hasAnyRecordedMetric(metric)).toBe(true);
   });
 
+  it('treats low-confidence clarification metric as recorded metric activity', () => {
+    const metric: MetricRecord = {
+      startedAt: Date.UTC(2026, 1, 1, 12, 0, 0),
+      workspaceRoot: '/workspace/repo',
+      trigger: 'manual',
+      lowConfidenceClarificationRate: 1,
+    };
```
[the remark this answers is not in the document]
Comment (jkordish) [reply test/metrics.test.ts]: Agreed. Updated in b664b4b: changed the test to use a casted `trigger: "other"` metric so `hasAnyRecordedMetric` depends on `lowConfidenceClarificationRate`.

File: docs/roadmap/v0.8.0-dynamic-percolation-gh-commands.sh [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  OUTPUT_DIR="$ROOT_DIR/docs/roadmap/.issue-body-staging/v0.8.0-$timestamp"
+  mkdir -p "$OUTPUT_DIR"
+fi
+
+extract_section_body() {
+  local heading="$1"
+  awk -v h="$heading" '
+    $0 == h { in_section=1; next }
+    in_section && /^### / { exit }
+    in_section { print }
```
[the remark this answers is not in the document]
Comment (jkordish) [reply docs/roadmap/v0.8.0-dynamic-percolation-gh-commands.sh]: Fixed in 921186e. `extract_section_body` now uses `grep -nxF` to locate the heading line number and then does numeric `awk` range extraction, avoiding interpolation of heading text into awk matching logic.

File: src/extension.ts [hunk 3]
```diff
@@ -2424,11 +2424,18 @@ async function presentSummary(
   }
 
   const percolationMode = resolveCompanionRuntimeMode(config);
+  const notificationPrimary = rankPercolationForSummary(summary, percolationMode, {
+    context,
+    workspaceRoot,
+  }).primary;
+  recordLowConfidenceClarificationRate(summary, notificationPrimary);
```
[the remark this answers is not in the document]
Comment (jkordish) [reply src/extension.ts]: Addressed in 921186e. `recordLowConfidenceClarificationRate` now runs only after suppression gating passes in `presentSummary`, so suppressed low-confidence sessions are no longer counted as clarification selections.
